v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/datastream/ShadowTableReadUtils.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.datastream;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.Date;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.teleport.v2.spanner.ddl.Column;
+import com.google.cloud.teleport.v2.spanner.ddl.Ddl;
+import com.google.cloud.teleport.v2.spanner.ddl.Table;
+import com.google.cloud.teleport.v2.spanner.type.Type;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ShadowTableReadUtils {
+  // TODO: After beam release, use the latest client lib version which supports setting lock
+  // hints via the read api. SQL string generation should be removed.
+  public static Statement generateShadowTableReadSQL(
+      String shadowTable, List<String> readColumnList, Key primaryKey, Ddl shadowTableDdl) {
+    String columnNames = String.join(", ", readColumnList);
+    // TODO: Handle json type as PKs.
+    String whereClause =
```

Comment:
Is this syntax same for PG dialect? If no, can you please add a TODO here to support PG dialect?

Reply:
No this won't work for PG dialect, but this is anyways an intermin solution. THe longer term is to use the API which is dialect agnostic